services/cloudfront/src/main/java/software/amazon/awssdk/services/cloudfront/CloudFrontUtilities.java

```diff
@@ -0,0 +1,264 @@
+            throw SdkClientException.create("Could not sign custom policy cookie", e);
+        }
+    }
+
+    /**
+     * Generate a policy document that describes custom access permissions to
+     * apply via a private distribution's signed URL.
+     *
+     * @param resourceUrl
+     *            The HTTP/S resource path that restricts which distribution and
+     *            S3 objects will be accessible in a signed URL, i.e.,
+     *            <tt>"https://" + distributionName + "/" + objectKey</tt> (may
+     *            also include URL parameters). The '*' and '?' characters can
+     *            be used as a wildcards to allow multi-character or single-character
+     *            matches respectively:
+     *            <ul>
+     *            <li><tt>*</tt> : All distributions/objects will be accessible</li>
+     *            <li><tt>a1b2c3d4e5f6g7.cloudfront.net/*</tt> : All objects
+     *            within the distribution a1b2c3d4e5f6g7 will be accessible</li>
+     *            <li><tt>a1b2c3d4e5f6g7.cloudfront.net/path/to/object.txt</tt>
+     *            : Only the S3 object named <tt>path/to/object.txt</tt> in the
+     *            distribution a1b2c3d4e5f6g7 will be accessible.</li>
+     *            </ul>
+     * @param activeDate
+     *            An optional UTC time and date when the signed URL will become
+     *            active. If null, the signed URL will be active as soon as it
+     *            is created.
+     * @param expirationDate
+     *            The UTC time and date when the signed URL will expire. REQUIRED.
+     * @param limitToIpAddressCidr
+     *            An optional range of client IP addresses that will be allowed
+     *            to access the distribution, specified as a CIDR range. If
+     *            null, the CIDR will be omitted and any client will be
+     *            permitted.
+     * @return A policy document describing the access permission to apply when
+     *         generating a signed URL.
+     */
+    public static String buildCustomPolicyForSignedUrl(String resourceUrl,
+                                                       Instant activeDate,
+                                                       Instant expirationDate,
+                                                       String limitToIpAddressCidr) {
+        if (expirationDate == null) {
+            throw SdkClientException.create("Expiration date must be provided to sign CloudFront URLs");
+        }
+        if (resourceUrl == null) {
+            resourceUrl = "*";
+        }
+        return SigningUtils.buildCustomPolicy(resourceUrl, activeDate, expirationDate, limitToIpAddressCidr);
+    }
+
+    /**
+     * Creates a private key from the file given, either in RSA private key
+     * (.pem) or pkcs8 (.der) format. Other formats will cause an exception to
+     * be thrown.
+     */
+    public static PrivateKey loadPrivateKey(File privateKeyFile) throws InvalidKeySpecException, IOException {
```

Comment:
Should this be private? Our API could take both privateKey(PrivateKey) and privateKey(Path) so that they don't have to find this loadPrivateKey method themselves when all they have is a file.

Also:
1. We shouldn't throw checked exceptions.
2. We should accept Path, not File
3. We should be clear that we're choosing pem/der based on the file name. Not anything more intelligent.

Reply:
moved to internal SigningUtils class.
updated to throw Exception.
updated to Path.
updated javadoc.

Reply:
In the builder, we could just have separate methods for each type?

```
public Builder privateKey(PrivateKey key) {
    this.privateKey = key;
    return this;
}

public Builder privateKey(Path keyFile) {
    this.privateKey = loadPrivateKey(keyFile);
    return this;
}
```

Reply:
added separate methods in request builder